### DISCUSSION TITLE
How to path aliases to work in a monorepo

### DISCUSSION THREAD

**Discussion by kharann** [trpc/trpc#3163] [Q&A]
Hi! I have a typescript monorepo where my backend uses path aliases (in tsconfig). It works completely fine running the backend.

However, the frontend breaks, which I assume is due to importing AppRouter type from the backend and since the frontend doesn't have the aliases defined in the backend, it just breaks. I would love to use path aliases in my backend and frontend as it makes the imports look cleaner.

**Comment by walleXD** [comment]
Depends entirely on your application stack and build toolchain.

In terms of exposing types from your pkg / api, you can point your package.json to it.

If you are exposing it directly, for expample `api` package, then in the `package.json` point the main and types keys to the desired file. For example:

```json
{
  "types": "./src/index.ts",
  "main": "./src/index.ts"
}
```

If you want to expose it from a subpath, for example `web/api`, then it's a little bit unintuitive but not too bad:
```json
  "typesVersions": {
    "*": {
      "api": [
        "./src/server/index.ts"
      ]
    }
  },
  "exports": {
    "./api": "./src/server/index.ts"
  }
```

**Reply by kharann** [reply]
More information:

My repository consists of `apps` and `packages`. Apps are runnable applications, while packages can be imported

- My server runs on `apps/api`, while the frontend runs on `apps/web`. 
- `apps/api` uses path aliases such as `@/trpc` and exports the AppRouter
- `apps/web` imports the AppRouter

What is index.ts in your example? Where I export AppRouter type? Thanks for the help btw! :D

**Reply by ghost** [reply]
> More information:
> 
> My repository consists of `apps` and `packages`. Apps are runnable applications, while packages can be imported
> 
> - My server runs on `apps/api`, while the frontend runs on `apps/web`. 
> - `apps/api` uses path aliases such as `@/trpc` and exports the AppRouter
> - `apps/web` imports the AppRouter
> 
> What is index.ts in your example? Where I export AppRouter type? Thanks for the help btw! :D 

I guess that you just create an `index.ts` inside your `packages/trpc` root, from which you can export the `AppRouter` type.

**Reply by ghost** [reply]
> > More information:
> > 
> > My repository consists of `apps` and `packages`. Apps are runnable applications, while packages can be imported
> > 
> > - My server runs on `apps/api`, while the frontend runs on `apps/web`. 
> > - `apps/api` uses path aliases such as `@/trpc` and exports the AppRouter
> > - `apps/web` imports the AppRouter
> > 
> > What is index.ts in your example? Where I export AppRouter type? Thanks for the help btw! :D 
> 
> I guess that you just create an `index.ts` inside your `packages/trpc` root, from which you can export the `AppRouter` type.

Might need `next-transpile-mofules` configured in your `next.config.js` as well.

**Reply by kharann** [reply]
> > More information:
> > My repository consists of `apps` and `packages`. Apps are runnable applications, while packages can be imported
> > 
> > * My server runs on `apps/api`, while the frontend runs on `apps/web`.
> > * `apps/api` uses path aliases such as `@/trpc` and exports the AppRouter
> > * `apps/web` imports the AppRouter
> > 
> > What is index.ts in your example? Where I export AppRouter type? Thanks for the help btw! :D
> 
> I guess that you just create an `index.ts` inside your `packages/trpc` root, from which you can export the `AppRouter` type.

Problem is that I'm not running trpc as a separate package, but rather as a part of the API app.

**Comment by kharann** [comment]
Is it perhaps possible to ignore types from the API when building? As they are covered in the own pipeline
